selenium/scripts/check_endpoints.pl

```diff
@@ -0,0 +1,106 @@
+#!/usr/local/bin/perl
+
+=head1 LICENSE
+
+Copyright [2009-2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 DESCRIPTION
+
+This script checks all endpoints in the REST server to see if they load 
+without error. It doesn't check *which* endpoints are present, or that 
+the endpoints return useful data, only that they don't fail.
+
+=head1 USAGE
+
+Download Selenium server standalone (requires version <= 2.52, get it here http://selenium-release.storage.googleapis.com)
+
+Run the server:
+java -jar selenium-server-standalone-2.52.0.jar
+
+Run the checker: 
+perl selenium/check_endpoints.pl --host 127.0.0.1 http://rest.ensembl.org
+
+=cut
+
+use strict;
+use warnings;
+use Getopt::Long;
+use LWP::UserAgent;
+use Test::More;
+use Test::WWW::Selenium;
+
+GetOptions(
+  'host=s'    => \(my $host    = 'localhost'),
+  'port=i'    => \(my $port    = 4444),
+  'browser=s' => \(my $browser = '*googlechrome'),
+  'timeout=i' => \(my $timeout = 30000),
+);
+
+my $base_url = $ARGV[0] || die "Please supply a URL to test";
+
+my $selenium   = Test::WWW::Selenium->new(
+  host        => $host,
+  port        => $port,
+  browser     => $browser,  
+  browser_url => $base_url,
+  _ua         => LWP::UserAgent->new(keep_alive => 5, env_proxy => 1, agent => "$0 ")
+);
+
+$selenium->set_timeout($timeout);
+
+test_endpoints();
+done_testing();
+exit 0;
+
+
+sub test_endpoints {
+  ok $selenium->open($base_url), 'homepage';
+  foreach my $href (get_hrefs_like('/documentation/info')) {   
+    ok $selenium->open($href) && eval { page_loads_without_error() }, $href;

```

Comment:
Please add parentheses to all ok() calls so that it is visually unambiguous which arguments are passed where. This one in particular, while not particularly difficult to decode, is already packed enough with content to require a pause to mentally do so.

Reply:
seems subjective, but done anyway